### PR TITLE
Deadchat plays everything - A simple way for admins to let deadchat control things.

### PR DESCRIPTION
**Pull request by Timberpoes**
## About The Pull Request

Right now we have a deadchat_control component, however basic setup for this component is a nightmare for admins in-game. As a result, we only realistically have deadchat controlled singulos and deadchat_plays_goose.

Adds a bunch of code improvements to the deadchat control component including a subtype that comes pre-baked with cardinal movements and is restricted to /atom/movable and improved code documentation.

When the `deadchat_control` component is removed, there's now proper cleanup which should restore the atom's state to a sane default.

The component hooks into examining and will display available commands and the remaining cooldown. There's also a host of other code improvements in the user-feedback aisle.

This is also added to VV in a toggle-style manner, allowing the deadchat component to be added and/or removed with an associated alert box to allow the admin to confirm what they're about to do.

Huge thanks to @Qustinnus for spending time trying to explain how to appropriately leverage components and implement this properly, as well as implementing a new signal to allow my component to hook into VV topic calls as necessary.

## Why It's Good For The Game

Deadchat plays everything.

## Changelog
:cl:
admin: Admins may now call the deadchat_plays proc on any movable atom in the game, allowing deadchat to move it around with zero effort. They should now call the generic deadchat_plays instead of deadchat_plays_goose to turn Birdboat into a deadchat controlled vomit goose.
admin: The VV dropdown now contains an option to enable or disable deadchat control. This comes with an alert confirmation box to allow you to back out and chat messages to tell you the result.
refactor: There's now a lot more feedback when deadchat is controlling something, including telling you when command inputs are successful, telling you the remaining cooldown until you can input new commands when it fails and additional examine text detailling the control mode, commands and overall cooldown.
/:cl:

### CODE REVIEW

**Comment by IndieanaJones** [comment]
Is there any way to have your inputs sent through using movement keys instead of saying the direction?  I think it'd make the experience a lot better.

**Comment by Jared-Fogle** [comment]
My question is why this is a better idea than just making `deadchat_controlled` default to movement, which means admins can just do Add Component/Element --> deadchat_controlled? It's not like this isn't a sane default.

At the same time, I don't care too much since you've integrated this proc into code actually used in game, so meh.

**Comment by Timberpoes** [comment]
Big chunk of additional changes. Code docs. Refactored areas of code. Created a component subtype with pre-baked movement commands that is incompatible with non-movable atoms.

**Comment by davethwave** [comment]
Now we just need the *say* command for the ghosts set up with a cooldown and then we can do some fun stuff like attaching this to a crewmember and play out the exorcist on station.

**Comment by TheVekter** [comment]
> 
> 
> Now we just need the _say_ command for the ghosts set up with a cooldown and then we can do some fun stuff like attaching this to a crewmember and play out the exorcist on station.

Can't imagine that happening. We had to remove ghosts rotating chairs because it was being used as a fucking divining system. Ghosts giving info to living players is a bad idea, even if an admin has to intervene to make it happen.

**Comment by Timberpoes** [comment]
It's ready for review and merge or for some maintainer to close it as far too much power for the hands of a simple admin to bestow upon a thing in the game world with such ease and simplicity.

File: code/game/atoms_movable.dm [hunk 1]
```diff
@@ -1041,7 +1041,43 @@
...
+	return AddComponent(/datum/component/deadchat_control/cardinal_movement, mode, list(), cooldown)
+
+/atom/movable/vv_get_dropdown()
+	. = ..()
+	VV_DROPDOWN_OPTION(VV_HK_DEADCHAT_PLAYS, "Start/Stop Deadchat Plays")
+
+/atom/movable/vv_do_topic(list/href_list)
+	. = ..()
+
+	if(!.)
```
Comment (AnturK) [inline code/game/atoms_movable.dm:1062]: From quick glance this isn't how return value was used in this proc before ?

**Review by AnturK — COMMENTED** [review]
Conflicts

File: code/game/atoms_movable.dm (hunk 1 above)
Comment (Timberpoes) [reply code/game/atoms_movable.dm:1062]: Feast your eyes on this monstrosity

```
/obj/vv_do_topic(list/href_list)
	if(!(. = ..()))
		return
```

I think this is code I borrowed from Floyd. I can change it to be similar to the other vv_do_topic overrides

Comment (Qustinnus) [reply code/game/atoms_movable.dm:1063]: I saw it in one place and there's a reason for it but I cant remember why.

Comment (Qustinnus) [reply code/game/atoms_movable.dm:1063]: Oh wait I remember, if you don't do this double popups can occur, If the signal sends a response aka COMPONENT_VV_HANDLED, the parent returns TRUE, if this is the case, we shouldn't do anything else, else you get the case where clicking on the add_deadchat will popup twice.

Comment (AnturK) [reply code/game/atoms_movable.dm:1063]: Not a fan how this is inconsistent between vv_do_topic implementations, but i guess that's for another PR

**Pull request merged**